### ISSUE TITLE
WSL2 failed to run, no response from virtual machine or container

### ISSUE THREAD

**Issue by wangfys** [microsoft/WSL#7351]
### Windows Build Number

Microsoft Windows [Version 10.0.19043.1151]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16

### Other Software

__

### Repro Steps

Just run 'wsl' command in PowerShell or create a new tab of Ubuntu in PowerShell.

### Expected Behavior

The WSL2 should run normally.

### Actual Behavior

It says "没有收到虚拟机或容器的回应，操作超时 [已退出进程，代码为4294967295]".
The translation of the Chinese should be 'No response from virtual machine or container, operation timed out [Process exited, the code is 4294967295]'

### Diagnostic Logs

Currently, I'm using a Surface Pro X running Windows 10 on ARM and the PowerShell version is PowerShell 7.2.0-preview.9

This issue doesn't appear on another computer running normal Windows 10.

I have searched whether there are similar issues. There are three related issues but their solutions are all workarounds. The real problem hasn't been found.
#7012 shows exactly the same behavior, but the solution he/she found is that install Ubuntu-18.04 in the MS store and then his 20.04 works fine. 
#7332 shows a similar behavior, the solution he/she found is to uninstall Windows Update KB5005033. I tried but failed.
#7339 shows a similar behavior, the solution he/she found is to export the distro, unregister it and then import it again.

Basically, I guess that there is something wrong with a Windows Update. It results in that the system can't find the original WSL distro correctly. Install another distro or export+unregister+import accidentally fixes this problem.

**Comment by wangfys** [comment]
I tried to install Ubuntu-18.04 in the MS store. It successfully runs but my original Ubuntu is still in trouble.

So the newly installed Ubuntu-18.04 is WSL1 so it works fine.
When I tried to convert it into WSL2, it says 'No response from virtual machine or container, operation timed out' again.
And my Ubuntu in WSL2 failed to run

![image](https://user-images.githubusercontent.com/16982796/130780593-45a3e0e7-3da7-43ef-bbaa-0ce83c6e182f.png)

**Comment by CoskunSunali** [comment]
I am receiving the same error on Windows 11 22000.168 (kernel version 5.10.43)

P.S. It occasionally works, mostly doesn't.

```bash
Coskun@CS MINGW64 ~
$ ubuntu
The operation timed out because a response was not received from the virtual machine or container.
Segmentation fault
```

**Comment by wangfys** [comment]
I have used Feedback Hub to upload the logs. Here is the link: https://aka.ms/AAdr1sz

**Comment by florentchauveau** [comment]
I have the same issue with Windows 11 Insider build 10.0.22454.1000.

> The operation timed out because a response was not received from the virtual machine or container.

**Comment by maxboone** [comment]
> I have the same issue with Windows 11 Insider build 10.0.22454.1000.
> 
> Surface Pro X (SQ2)
> 
> > The operation timed out because a response was not received from the virtual machine or container.

Same issue, steps that didn't help:
- Disabling & re-enabling the features
- Uninstalling both wslg & kernel updates from software center and installing them using `wsl --update`
- Running `netcfg -d`
- Installing & registering a new distro (same error occurs)
- Invoking WSL with the `-e` flag to specify it to start `/bin/sh`
- Don't have anything running on port 53
- Restarting several services, `LxssManager`, `vmcompute`, `SharedAccess`
- Repairing system image with `sfc /scannow` and `dism /Online /Cleanup-Image /RestoreHealth`
- Setting a `$WINDOWS_HOME/.wslconfig`

I think I'm just going to do a re-install tbh, it seems like something is really up with the network interface between the Hyper-V guest services & wsl client side.

That said, I don't have the `vmsmp` service (even after above steps) which I saw in another issue - could that be related?

**Comment by readloud** [comment]
wsl --shutdown

**Comment by Arias1101** [comment]
wsl --shutdown
works for me, thanks !

**Comment by readloud** [comment]
👌

**Comment by khadeshyam** [comment]
wsl: Failed to launch the localhost relay process. Error: The operation timed out because a response was not received from the virtual machine or container.
The operation timed out because a response was not received from the virtual machine or container.

Error code: Wsl/Service/CreateInstance/CreateVm/HCS_E_CONNECTION_TIMEOUT

[process exited with code 4294967295 (0xffffffff)]
You can now close this terminal with Ctrl+D, or press Enter to restart.

**Comment by khadeshyam** [comment]
![Screenshot (105)](https://github.com/microsoft/WSL/assets/96071288/f76eddbd-6cff-4e12-8038-c48eb8e579c3)
 I am getting this error can someobe help

**Comment by dhruvmsheth** [comment]
finally fixed it. nothing in the comments above helped me. I navigated to the .wslconfig files placed in `/Users/<username>/` and opened .wslconfig file with a text editor and removed a line which redirected the kernel to a non-existent directory. Make sure the config file only says `[wsl2]`

**Comment by nevernotmove** [comment]
I had all kinds of WSL errors. What finally seemed to fix it was that for some reason the memory size and swap size in the Windows WSL settings where set to a miniscule size. After resetting them, it seems to work fine right now.
